src/main/java/hudson/plugins/release/pipeline/ReleaseStep.java

```diff
@@ -0,0 +1,160 @@
+package hudson.plugins.release.pipeline;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import javax.annotation.Nonnull;
+
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.jenkinsci.plugins.workflow.util.StaplerReferer;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.DoNotUse;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerRequest;
+
+import com.google.common.collect.ImmutableSet;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.AutoCompletionCandidates;
+import hudson.model.BuildableItem;
+import hudson.model.BuildableItemWithBuildWrappers;
+import hudson.model.ItemGroup;
+import hudson.model.Job;
+import hudson.model.ParameterDefinition;
+import hudson.model.ParameterValue;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.release.ReleaseWrapper;
+import jenkins.model.Jenkins;
+import jenkins.model.ParameterizedJobMixIn;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+/**
+ * Pipeline step implementation for Release plugin.
+ *
+ * @author Alexey Merezhin
+ * @since 2.7
+ */
+public class ReleaseStep extends Step {
+    private static final Logger LOGGER = Logger.getLogger(ReleaseStep.class.getName());
+
+    private String job;
+    @Nonnull
+    private List<ParameterValue> parameters;
+
+    @DataBoundConstructor
+    public ReleaseStep(String job) {
+        this.job = job;
+        parameters = new ArrayList<>();
+    }
+
+    public String getJob() {
+        return job;
+    }
+
+    public void setJob(String job) {
+        this.job = job;
+    }
+
+    public List<ParameterValue> getParameters() {
+        return parameters;
+    }
+
+    @DataBoundSetter
+    public void setParameters(List<ParameterValue> parameters) {
+        if (parameters == null) {
+            this.parameters = new ArrayList<>();
+        } else {
+            this.parameters = parameters;
+        }
+    }
+
+    @Override
+    public StepExecution start(StepContext stepContext) throws Exception {
+        return new ReleaseStepExecution(stepContext, this);
+    }
+
+    @Extension
+    public static class DescriptorImpl extends StepDescriptor {
+
+        @Override
+        public String getFunctionName() {
+            return "release";
```

Comment:
I talked to @kohsuke about this and we both thought this seemed too generic but I can't think of a better alternative right now. Any thoughts?